backend/typescript/services/implementations/reviewService.ts

```diff
@@ -189,7 +192,119 @@ class ReviewService implements IReviewService {
+  // if deleteReview is called without parameter txn, it runs in a try catch block
+  // otherwise, deleteReview should be called inside a try catch block with t as parameter
+  async deleteReview(id: string, txn?: Transaction): Promise<void> {
+    if (txn) {
+      const reviewToDelete = await PgReview.findByPk(id, {
+        transaction: txn,
+        include: [{ all: true, nested: true }],
+      });
+
+      if (!reviewToDelete) {
+        throw new Error(`Review id ${id} not found`);
+      }
+
+      const allBookIds = reviewToDelete.books.map((book: PgBook) => book.id);
+      const deletePublishersAndAuthors = Promise.all(
+        reviewToDelete.books.map(async (book: PgBook) => {
+          // Delete authors (if necessary)
+          book.authors.forEach((author: PgAuthor) => {
+            PgBookAuthor.findAll({
+              where: { author_id: author.id },
+            }).then(async (ret: PgBookAuthor[]) => {
+              const bookAuthorIds = ret.map(
+                (bookAuthor: PgBookAuthor) => bookAuthor.book_id,
+              );
+              if (
+                bookAuthorIds.every((bookId) => allBookIds.includes(bookId))
+              ) {
+                // Delete author
+                await PgAuthor.destroy({
+                  where: { id: [author.id] },
+                });
+              }
+            });
+          });
+
+          // Delete publishers (if necessary)
+          book.publishers.forEach((publisher: PgPublisher) => {
+            PgBookPublisher.findAll({
+              where: { publisher_id: publisher.id },
+            }).then(async (ret: PgBookPublisher[]) => {
+              const bookPublisherIds = ret.map(
+                (bookPublisher: PgBookPublisher) => bookPublisher.book_id,
+              );
+              if (
+                bookPublisherIds.every((bookId) => allBookIds.includes(bookId))
+              ) {
+                // Delete publisher
+                await PgPublisher.destroy({
+                  where: { id: [publisher.id] },
+                });
+              }
+            });
+          });
+        }),
+      );
+
+      const deleteBooks = Promise.all(
+        reviewToDelete.books.map(async (book: PgBook) => {
+          // Delete book
+          PgBook.destroy({
+            where: { id: [book.id] },
+          }).then(() => {
+            // Delete series (if necessary)
+            if (book.series) {
+              PgBook.findAll({
+                where: { series_id: book.series.id },
+              }).then((ret: PgBook[]) => {
+                // We check for 0 because we deleted the book (can't be 1)
+                if (ret.length === 0) {
+                  // Delete series
+                  PgSeries.destroy({
+                    where: { id: [book.series.id] },
+                  });
+                }
+              });
+            }
+          });
+        }),
+      );
+
+      // Delete tags (if necessary)
+      const deleteTags = Promise.all(
```

Comment:
This may not relate to your ticket, but could you remove all references to Tags in the insert and delete methods? We're refactoring Tags rn and these changes might cause conflicts